### PR TITLE
relates to #1734: read resource

### PR DESCRIPTION
**Pull request by ivansenic**
**What this PR does**:
Implementation of the read documents resource for the V2. With the complete and extended API documentation, showcasing different examples. Tests to come in a follow-up PR.

**Which issue(s) this PR fixes**:
Relates to #1734

**Checklist**
- [ ] Rebase on  #1916 to resolve documentation conflicts

### CODE REVIEW

File: sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/ReadResource.java
Comment (EricBorczuk) [inline sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/ReadResource.java:392]: https://www.youtube.com/watch?v=rKTUAESacQM

**Review by EricBorczuk — APPROVED** [review]
Looks good to me!

File: sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/StargateDocsApi.java
Comment (tatu-at-datastax) [inline sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/StargateDocsApi.java:75]: blog -> blob

File: sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/ReadResource.java
Comment (tatu-at-datastax) [reply sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/ReadResource.java:168]: This is just my uninformed opinion, but I think increasing makes sense (not surprised there's "only 3?" question every now and then) and somehow feel 10 might be a good default; anything less seems unusual. 20 would work too.

File: sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/ReadResource.java
Comment (tatu-at-datastax) [inline sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/ReadResource.java:390]: Was curious about `? extends JsonNode` here: I am guessing that simple `Function<DRW<JsonNode>>` would work well unless there are places where one really wants `Function<DRW<ObjectNode>>` or such?
Might reduce need for casting too.

**Comment by EricBorczuk** [comment]
@ivansenic Just merged my write-path controller PR and rebased this on the v2 main branch!

File: sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/ReadResource.java
Comment (EricBorczuk) [reply sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/ReadResource.java]: Yea - as far as current usage, I don't see the need for `? extends JsonNode` - also I think in theory anything consuming the DocumentResponseWrapper doesn't need a more specific type than JsonNode

I changed this and will merge when CI passes!

**Comment by ivansenic** [comment]
@EricBorczuk @tatu-at-datastax  I'll cover all the comments you mentioned in the ticket that covers the tests..

File: sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/ReadResource.java
Comment (ivansenic) [reply sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/ReadResource.java:168]: Would keep 3 for now, but yes up for the discussion.. Main reason was not to load too many documents on the "default" page size..

Comment (tatu-at-datastax) [reply sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/ReadResource.java:168]: Right, I remember hearing that the low size was meant to limit disk access and in a way benefit users.
This might be something to talk with Lorina and/or Mark at some point. Maybe it is better to wait for user feedback before changing things.

**Pull request merged**
